snow/engine/snowman/transitive_test.go

```diff
@@ -2857,6 +2857,137 @@ func TestEngineVoteStallRegression(t *testing.T) {
 	require.Equal(choices.Rejected, rejectedChain[0].Status())
 }
 
+// When a voter is registered with multiple dependencies, the engine must not
+// execute the voter until all of the dependencies have been resolved; even if
+// one of the dependencies has been abandoned.
+func TestEngineEarlyTerminateVoterRegression(t *testing.T) {
```

Comment:
This is a regression testing the original pass of this PR in commit 1c1501dd60691d37d23d88f7ae8dc2b296b4b6fc. This regression was fixed inside this PR with commit e99eee0b3d832aec7f33b21bae67a1281351902a.

This bug never existed on the master branch, but it was a subtle enough of an issue that I felt it deserved an explicit test. (And clearly it is a mistake people could make... I made it.)

There is going to be a pass soon to cleanup all of these engine tests... But the first step of that is simplifying the engine and having better abstractions to test against.